core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Submission.java

```diff
@@ -23,62 +23,141 @@
-@JsonTypeInfo(use=JsonTypeInfo.Id.CLASS, include= JsonTypeInfo.As.PROPERTY, property="@class")
-@JsonIgnoreProperties({"securityProfile"})
-public interface Submission<I, O, R> extends Submittable<O>, SecurableByProfile, Nameable, Comparable {
+import java.io.Serializable;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+
+@Entity
+@Table(name = "Submission")
+public class Submission implements Comparable<Submission>, Serializable
+{
 
   public static final Long UNSAVED_ID = 0L;
 
-  @Deprecated
-  public Long getSubmissionId();
-  @Deprecated
-  public void setSubmissionId(Long submissionId);
-
-  public void setId(long id);
-
-  public void setName(String name);
-  public String getAlias();
-  public void setAlias(String alias);
-  public String getAccession();
-  public void setAccession(String accession);
-  public String getDescription();
-  public void setDescription(String description);
-  public String getTitle();
-  public void setTitle(String title);
-  public Date getCreationDate();
-  public void setCreationDate(Date creationDate);
-  public Date getSubmissionDate();
-  public void setSubmissionDate(Date submissionDate);
-  public boolean isVerified();
-  public void setVerified(boolean verified);
-  public boolean isCompleted();
-  public void setCompleted(boolean completed);
-  public void addSubmissionElement(I i);
-  public Set<Submittable<O>> getSubmissionElements();
-  public SubmissionActionType getSubmissionActionType();
-  public void setSubmissionActionType(SubmissionActionType submissionActionType);
-  public R submit(SubmissionManager<I, O, R> manager) throws SubmissionException;
-
-  Date getLastUpdated();
-  void setLastUpdated(Date lastUpdated);
+  private static final long serialVersionUID = 1L;
+
+  private String accession;
+  private String alias;
+  private boolean completed;
+
+  @Temporal(TemporalType.DATE)
+  private Date creationDate;
+
+  private String description;
+
+  @ManyToMany(targetEntity = Experiment.class)
+  @JoinTable(name = "Submission_Experiment", joinColumns = {
+      @JoinColumn(name = "submission_submissionId") }, inverseJoinColumns = {
+          @JoinColumn(name = "experiments_experimentId") })
+  private Set<Experiment> experiments = new HashSet<>();
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long submissionId = UNSAVED_ID;
+
+  @Temporal(TemporalType.DATE)
+  private Date submittedDate;
+
+  private String title;
+  private boolean verified;
+
+  @Override
+  public int compareTo(Submission t) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1210)